components/Countdown.js

```diff
@@ -0,0 +1,67 @@
+import React from "react";
+import PropTypes from "prop-types";
+import timediff from "timediff";
+
+class Countdown extends React.Component {
+  constructor(props) {
+    super(props);
+    this.updateCountdown = this.updateCountdown.bind(this);
+    this.dateToCountDownTo = props.initialDate;
+    this.currentDate = Date.now();
+    this.Difference = this.getDifferenceInDates();
+    this.state = {
+      days: this.Difference.days,
+      hours: this.Difference.hours,
+      minutes: this.Difference.minutes,
+      seconds: this.Difference.seconds,
+    };
+  }
+
+  componentDidMount() {
+    setInterval(this.updateCountdown, 1000);
```

Comment:
You forgot to clean up the interval in the `componentWillUnmount`. This will cause a memory leak...